src/rulesets/Base/arraymath.jl

```diff
@@ -268,3 +268,19 @@ function rrule(::typeof(-), x::AbstractArray)
     end
     return -x, negation_pullback
 end
+
+
+#####
+##### Addition (Multiarg `+`)
+#####
+
+function rrule(::typeof(+), arrs::AbstractArray...)
+    y = +(arrs...)
+    function add_pullback(dy)
+        return (
+            NoTangent(),
+            ntuple(i -> dy .* ones(eltype(arrs[i]), size(arrs[i])), length(arrs))...
```

Comment:
Is there a reason we are not using `fill` here instead?

Also, have you considered using thunks? https://juliadiff.org/ChainRulesCore.jl/dev/writing_good_rules.html#Use-Thunks-appropriately

Reply:
This seems complicated. Isn't the rule just like `dy -> (nothing, dy, dy, dy, ...)`?

Reply:
@mcabbott Indeed! I'll update it.

Reply:
Maybe one further question is how much you want to worry about cases like this, Zygote certainly does not:
```julia
julia> ones(3) + ones(3,1)
3×1 Matrix{Float64}:
 2.0
 2.0
 2.0

julia> Zygote.gradient(sum∘+, ones(3), ones(3,1))
(3×1 Fill{Float64}: entries equal to 1.0, 3×1 Fill{Float64}: entries equal to 1.0)
```